python/config_settings/config_settings.bzl

```diff
@@ -38,8 +38,10 @@ def construct_config_settings(name, python_versions):
     string_flag(
         name = "python_version",
         # TODO: The default here should somehow match the MODULE config
-        build_setting_default = python_versions[0],
-        values = sorted(allowed_flag_values),
+        # NOTE @aignas 2024-01-17: python_versions[0] as default makes the
+        # default wheel selection on bzlmod not work.
+        build_setting_default = "",
```

Comment:
@rickeylev, I have found that without this line the `multi_python_version` test for default PyPI binary selection would fail unless I have this. Not sure why this is the case. I think it is because `python_versions[0]` is `3.8` so I guess we should have a special value here, which an empty string is.

Reply:
That's weird.

python_versions is a list of full version strings, so it would be 3.8.X, not 3.8.

Is it because...do the multi-version code paths special case a falsey default value maybe? Maybe they're actually generating a no_match_error instead of //conditions:defualt entry, or something?

Reply:
I've no idea about this to be honest, if I set this to `python_versions[0]`, which is `3.8.x`, then the `websockets` version that we get in the test is for Python 3.8 and not 3.9 as it should be. Is there a way to somehow print the trace of how the select happens?

Reply:
I don't think so...

To make sure I understand...
* Setting it so 3.8.x failed
* Setting it to empty string passed
* Setting it to "PY3" passed

That seems even weirder. It doesn't make much sense, since this is the default for the flag, and only handful of things directly read that flag...

The only idea i have is something in the multi-pip-parse example has something like `select({":3.8.x": ..., "//conditions:default": ...})`. So an empty string and PY3 go down the default case.

What if there was a bug here all along? Like "3.8.1" was being set, but the select was looking for "3.8.2", and then somehow //conditions:default just ended up somewhere happy enough for tests to pass? And by changing the select to effectively match _both_ 3.8.1 and 3.8.2, it's now doing the _intended_ thing and showing some misconfiguration?